library/Notifications/Web/Control/SearchBar/ObjectSuggestions.php

```diff
@@ -231,10 +231,9 @@ protected function queryTags(Model $model, string $searchTerm): Query
 
     protected function matchSuggestion($path, $label, $searchTerm)
     {
-        if (preg_match('/[_.](id)$/', $path)) {
+        if (preg_match('/[_.](id)$/', $path, $matches)) {
             // Only suggest exotic columns if the user knows about them
-            $trimmedSearch = trim($searchTerm, ' *');
-            return substr($path, -strlen($trimmedSearch)) === $trimmedSearch;
+            return $matches[1] === trim($searchTerm, ' *');
```

Comment:
This makes it too easy to find such columns. The previous intent was to only show these if the user knows their exact path.

The previous condition however was indeed incorrect. I've attempted to fix this myself earlier and came up with this instead:

```
return substr($path, strrpos($path, '.') + 1) === $trimmedSearch;
```

Though, this was only a quick fix. Please check if this matches the original intent better.

Reply:
This works fine.